src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/SelectExpandNode.cs

```diff
@@ -84,24 +82,102 @@ public SelectExpandNode(IEdmStructuredType structuredType, ODataSerializerContex
         /// <param name="structuredType">The structural type of the resource that would be written.</param>
         /// <param name="model">The <see cref="IEdmModel"/> that contains the given structural type.</param>
         public SelectExpandNode(SelectExpandClause selectExpandClause, IEdmStructuredType structuredType, IEdmModel model)
-            : this(selectExpandClause, structuredType, model, false)
+            : this()
         {
+            Initialize(selectExpandClause, structuredType, model, false);
         }
 
         /// <summary>
-        /// Creates a new instance of the <see cref="SelectExpandNode"/> class describing the set of structural properties,
-        /// nested properties, navigation properties, and actions to select and expand for the given <paramref name="selectExpandClause"/>.
+        /// Gets the list of EDM navigation properties to be expand referenced in the response.
+        /// keeping this is only for non-breaking changes, This should be replaced by "ReferencedProperties" later.
         /// </summary>
```

Comment:
should we mark as obsolete?

Reply:
Yes. We can. However, I don't want to introduce breaking changes. Let's leave the change when we do V8?

---
In reply to: [355518435](https://github.com/OData/WebApi/pull/1941#discussion_r355518435) [](ancestors = 355518435)

Reply:
Would rather mark as obsolete now, and remove when we do V8.

---
In reply to: [355615209](https://github.com/OData/WebApi/pull/1941#discussion_r355615209) [](ancestors = 355615209,355518435)

Reply:
Done

---
In reply to: [356250942](https://github.com/OData/WebApi/pull/1941#discussion_r356250942) [](ancestors = 356250942,355615209,355518435)

Reply:
Doesn't look this one was marked obsolete, only SelectedComplexProperties?

---
In reply to: [356251862](https://github.com/OData/WebApi/pull/1941#discussion_r356251862) [](ancestors = 356251862,356250942,355615209,355518435)

Reply:
We talked this? maybe i missed that. Updating...

---
In reply to: [356322937](https://github.com/OData/WebApi/pull/1941#discussion_r356322937) [](ancestors = 356322937,356251862,356250942,355615209,355518435)

Reply:
Not a big deal, but as long as we are marking the old pattern as obsolete we might as well include this one.